deployment/deploy/104_MocIntegrationWithIntermediaryDllr.ts

```diff
@@ -0,0 +1,71 @@
+import { ethers, network } from "hardhat";
+import { DeployFunction } from "hardhat-deploy/types";
+import { upgradeWithTransparentUpgradableProxy } from "../helpers/deployment";
+
+const func: DeployFunction = async ({ deployments, getNamedAccounts }) => {
```

Comment:
i like the design approach to differentiate at deployment level not changing MocIntegration contract )